pkg/provisioner/ironic/raid.go

```diff
@@ -273,18 +273,21 @@ func BuildRAIDCleanSteps(raidInterface string, target *metal3v1alpha1.RAIDConfig
 	return
 }
 
-func checkRAIDConfigure(raidInterface string, raid *metal3v1alpha1.RAIDConfig) error {
+// CheckRAIDInterface checks the current RAID interface against the requested configuration
+func CheckRAIDInterface(raidInterface string, target *metal3v1alpha1.RAIDConfig) error {
```

Comment:
I'm likely overthinking about it, but instead of making public directly this method, could be worth maybe exposing it through a more generic validation interface? I'm wondering if we could have similar needs in future (still in the validation area), and providing a single entry point for the external user may avoid exposing directly implementation details, and possibly help the maintenance.

Reply:
Maybe, but let's save the pressing problem now and think about refactoring later? This change (previously a part of the software RAID fix) has been blocking the RAID work on the terraform side for months already.